### PR TITLE
Make flavour copy-in relative to flavour dir

### PR DESCRIPTION
**Pull request by grembo**
This is acomplished by changing directories (seemed least
intrusive).

This allows having:
```
copy-in -s dns.d/resolver.conf -d /etc
```

### CODE REVIEW

**Comment by pizzamig** [comment]
I think that this implementation is too broad
* changing folder should be restricted to copy-in (it's affecting `mount-in` paths for no reason)
* changing folder should be executed iff file/directory is in the flavour_dir path and not in the current folder, to ensure retro-compatibility

One possible simple solution is to run the copy-in command as:
```
$( cd $_flv_dir && pot-cmd copy-in ...)
```
or re-using your implementation, that seems more solid than a sub-shell.

As for the other PR, an entry in the CHANGELOG file, is welcome ;)

**Comment by grembo** [comment]
> I think that this implementation is too broad
> 
> * changing folder should be restricted to copy-in (it's affecting `mount-in` paths for no reason)
> * changing folder should be executed iff file/directory is in the flavour_dir path and not in the current folder, to ensure retro-compatibility
> 
> One possible simple solution is to run the copy-in command as:
> 
> ```
> $( cd $_flv_dir && pot-cmd copy-in ...)
> ```
> 
> or re-using your implementation, that seems more solid than a sub-shell.
> 
> As for the other PR, an entry in the CHANGELOG file, is welcome ;)

I get the point about only changing dir on copy-in (even though one could argue that running *all* command while being in flavour dir could actually add some value).

I'm not a fan of changing behavior based on which directory the caller is in though - this could be highly confusing (so check first if relative now, otherwise chdir to flavourdir).

So maybe adding a new command in there would make more sense - e.g., copy-on-flv or something like this. This way existing behavior is preserved and new behavior will be predictable.

Thoughts?

**Comment by grembo** [comment]
Changed the pull request to implement what I suggested.

**ALSO** (which I forgot to mention in the commit message):

Change flavour behaviour to fail if pot-cmd fails. I'm not sure if the previous behaviour was intentional, but while testing the feature changes I was quite irritated by the fact that a missing source file in copy-in wouldn't stop creating the flavour (it would just show an error message, but continue just fine). This might break existing setups, but in terms of having repeatable/stable results I think this makes more sense.

```diff
-if ! pot-cmd $line -p "$_pname" ; then
- 	return 1
-fi
+pot-cmd $line -p "$_pname"
```

File: share/pot/common-flv.sh [hunk 1]
```diff
@@ -102,9 +103,26 @@ _exec_flv()
...
+					line=$(echo "$line" | sed "s/^copy-in-flv/copy-in/")
+					if ! pot-cmd $line -p "$_pname" ; then
+						return 1
+					fi
+					if ! cd "$_previous_pwd"; then
+						_error "Can't chdir to previous pwd $_previous_pwd"
+						return 1
+					fi
 				else
 					# shellcheck disable=SC2086
```
Comment (grembo) [inline share/pot/common-flv.sh:122]: This shellcheck seems unnecessary (even though I don't really get why). Shellcheck isn't complaining.

File: share/pot/common-flv.sh [hunk 2]
```diff
@@ -102,9 +103,26 @@ _exec_flv()
...
-					pot-cmd $line -p "$_pname"
+					if ! pot-cmd $line -p "$_pname" ; then
```
Comment (grembo) [inline share/pot/common-flv.sh:123]: This shellcheck seems unnecessary (even though I don't really get why). Shellcheck isn't complaining though (that's why I don't have it above)

**Pull request merged**